src/main/java/core/basesyntax/Dao.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface Dao<T> {
```

Comment:
```suggestion
public interface GenericDao<T> {
```
nice idea to create a generic dao